tests/unit/cli_test.py

```diff
@@ -61,30 +61,36 @@ def test_project_name_from_environment_new_var(self):
             project_name = command.get_project_name(None)
         self.assertEquals(project_name, name)
 
-    def test_filename_check(self):
-        self.assertEqual('docker-compose.yml', get_config_filename_for_files([
+    def subtest_filename_check(self, predicate):
+        self.assertEqual('docker-compose.yml', predicate([
             'docker-compose.yml',
             'docker-compose.yaml',
             'fig.yml',
             'fig.yaml',
         ]))
 
-        self.assertEqual('docker-compose.yaml', get_config_filename_for_files([
+        self.assertEqual('docker-compose.yaml', predicate([
             'docker-compose.yaml',
             'fig.yml',
             'fig.yaml',
         ]))
 
-        self.assertEqual('fig.yml', get_config_filename_for_files([
+        self.assertEqual('fig.yml', predicate([
             'fig.yml',
             'fig.yaml',
         ]))
 
-        self.assertEqual('fig.yaml', get_config_filename_for_files([
+        self.assertEqual('fig.yaml', predicate([
             'fig.yaml',
         ]))
 
-        self.assertRaises(ComposeFileNotFound, lambda: get_config_filename_for_files([]))
+        self.assertRaises(ComposeFileNotFound, lambda: predicate([]))
+
+    def test_filename_check(self):
+        self.subtest_filename_check(predicate=get_config_filename_for_files)
+
+    def test_filename_check_in_subdir(self):
+        self.subtest_filename_check(predicate=lambda x: get_config_filename_for_files(x, subdir=True))
```

Comment:
This business of passing a predicate in is a bit confusing. Could the whole thing be refactored to be clearer?